mod/caching/set/type_plus_right/metric/all_values.rb

```diff
@@ -2,38 +2,18 @@
 # company => [{ year => ..., value => ...}, ... ]
 include Card::CachedCount
 
-def cached_count
-  cached_count_card.content
-end
+# recount # of values for a metric when ...
 
-# expire if value was touched
-expired_cached_count_cards(
-  set: TypePlusRight::MetricValue::Value
-) do |changed_card|
-  changed_card.metric_card.fetch(trait: :all_values)
+# ... a +value is changed
+recount_trigger(TypePlusRight::MetricValue::Value) do |changed_card|
+  changed_card.metric_card.fetch trait: :all_values
 end
 
-# expire if name was changed (that includes for example a year change)
-expired_cached_count_cards set: Type::MetricValue,
-                           changed: :name do |changed_card|
+# ... a Metric Value (type) is renamed
+recount_trigger Type::MetricValue, changed: :name, on: :update do |changed_card|
   changed_card.metric_card.fetch(trait: :all_values)
 end
 
-# expire if company name was changed
-ensure_set { Type::MetricCompany }
-expired_cached_count_cards set: Type::MetricCompany,
-                           changed: :name do |changed_card|
-  metrics = Card.search type_id: MetricID, right_plus: changed_card.name
-  metrics.map do |metric|
-    metric.fetch trait: :all_values
-  end
-end
-
-# initialize cached values hash if new metric is created
-expired_cached_count_cards set: Type::Metric, on: :create do |changed_card|
-  changed_card.fetch trait: :all_values
-end
```

Comment:
I think deleting `def cached_count` above means this will be generated lazily, no?